sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -27,13 +27,21 @@ import org.apache.spark.sql.catalyst.util.truncatedString
 import org.apache.spark.sql.execution.exchange.ShuffleExchangeExec
 import org.apache.spark.sql.execution.metric.{SQLShuffleReadMetricsReporter, SQLShuffleWriteMetricsReporter}
 
+/**
+ * The operator takes limited number of elements from its child operator.
+ */
+trait LimitExec extends UnaryExecNode {
+  /** Number of element should be taken from child operator */
+  def limit: Int
+}
+
 /**
  * Take the first `limit` elements and collect them to a single partition.
  *
  * This operator will be used when a logical `Limit` operation is the final operator in an
  * logical plan, which happens when the user is collecting results back to the driver.
  */
-case class CollectLimitExec(limit: Int, child: SparkPlan) extends UnaryExecNode {
+case class CollectLimitExec(limit: Int, child: SparkPlan) extends LimitExec {
```

Comment:
I'm not 100% sure how Scala works here, but is `limit: Int` going to implement `def limit: Int` in the trait? I assume so or else this won't compile, so OK.

Reply:
> is `limit: Int` going to implement `def limit: Int` in the trait?

Yes, `limit: Int` is actually `val limit: Int` in case classes, and `val` can override `def`.